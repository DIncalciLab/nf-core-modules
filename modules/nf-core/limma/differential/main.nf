process LIMMA_DIFFERENTIAL {
    tag "$meta"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/bioconductor-edger_bioconductor-ihw_bioconductor-limma_r-dplyr_r-readr:7fc48564d286c1c6' :
        'community.wave.seqera.io/library/bioconductor-edger_bioconductor-ihw_bioconductor-limma_r-dplyr_r-readr:edea0f9fbaeba3a0' }"

    input:
    tuple val(meta), val(contrast_variable), val(reference), val(target)
    tuple val(meta2), path(samplesheet), path(intensities)

    output:
    tuple val(meta), path("*.limma.results.tsv")          , emit: results
    tuple val(meta), path("*.limma.mean_difference.png")  , emit: md_plot
    tuple val(meta), path("*.MArrayLM.limma.rds")         , emit: rdata
    tuple val(meta), path("*.limma.model.txt")            , emit: model
    tuple val(meta), path("*.R_sessionInfo.log")          , emit: session_info
    tuple val(meta), path("*.normalised_counts.tsv")      , emit: normalised_counts, optional: true
    path "versions.yml"                                   , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    template 'limma_de.R'
}
