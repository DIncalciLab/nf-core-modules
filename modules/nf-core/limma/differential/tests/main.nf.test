nextflow_process {

    name "Test Process LIMMA_DIFFERENTIAL"
    script "../main.nf"
    process "LIMMA_DIFFERENTIAL"

    tag "modules"
    tag "modules_nfcore"
    tag "limma"
    tag "limma/differential"

    test("test_limma_differential - exclude_samples") {

        config "./nextflow.exclude_samples.config"

        when {
            process {
                """
                input[0] = Channel.of(['id': 'test', 'variable': 'treatment', 'reference': 'hND6', 'target': 'mCherry'])
                        .map{tuple(it, it.variable, it.reference, it.target)}
                input[1] = Channel.of([[id:'test'],
                                      file(params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/SRP254919.samplesheet.csv', checkIfExists: true),
                                      file(params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/SRP254919.salmon.merged.gene_counts.top1000cov.tsv', checkIfExists: true)])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_limma_differential - subset_to_contrast") {

        config "./nextflow.subset_to_contrast.config"

        when {
            process {
                """
                input[0] = Channel.of(['id': 'test', 'variable': 'treatment', 'reference': 'hND6', 'target': 'mCherry'])
                        .map{tuple(it, it.variable, it.reference, it.target)}
                input[1] = Channel.of([[id:'test'],
                                      file(params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/SRP254919.samplesheet.csv', checkIfExists: true),
                                      file(params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/SRP254919.salmon.merged.gene_counts.top1000cov.tsv', checkIfExists: true)])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_limma_differential - voom_blocking") {

        config "./nextflow.voom_blocking.config"

        when {
            process {
                """
                input[0] = Channel.of(['id': 'test', 'variable': 'treatment', 'reference': 'hND6', 'target': 'mCherry'])
                        .map{tuple(it, it.variable, it.reference, it.target)}
                input[1] = Channel.of([[id:'test'],
                                      file(params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/SRP254919.samplesheet.csv', checkIfExists: true),
                                      file(params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/SRP254919.salmon.merged.gene_counts.top1000cov.tsv', checkIfExists: true)])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_limma_differential - voom") {

        config "./nextflow.voom.config"

        when {
            process {
                """
                input[0] = Channel.of(['id': 'test', 'variable': 'treatment', 'reference': 'hND6', 'target': 'mCherry'])
                        .map{tuple(it, it.variable, it.reference, it.target)}
                input[1] = Channel.of([[id:'test'],
                                      file(params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/SRP254919.samplesheet.csv', checkIfExists: true),
                                      file(params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/SRP254919.salmon.merged.gene_counts.top1000cov.tsv', checkIfExists: true)])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_limma_differential - stub") {

        options "-stub"

        when {
            process {                """
                input[0] = Channel.of(['id': 'test', 'variable': 'treatment', 'reference': 'hND6', 'target': 'mCherry'])
                        .map{tuple(it, it.variable, it.reference, it.target)}
                input[1] = Channel.of([[id:'test'],
                                      file(params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/SRP254919.samplesheet.csv', checkIfExists: true),
                                      file(params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/SRP254919.salmon.merged.gene_counts.top1000cov.tsv', checkIfExists: true)])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}