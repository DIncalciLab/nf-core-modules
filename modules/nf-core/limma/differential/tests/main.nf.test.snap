{
    "test_limma_differential - voom_blocking": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.results.tsv:md5,315c3bb68f55ab03b552111fd37af2ca"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.mean_difference.png:md5,2fb1e9594c24c0ab21d228e267cb3dcf"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.MArrayLM.limma.rds:md5,dea2517c23b7fa990bbc6eb6add2d00e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.model.txt:md5,3b96713b4e3f027b0347859f02a9038d"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.R_sessionInfo.log:md5,33a080b950790446e947baa6ddb7051e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.normalised_counts.tsv:md5,2aa4880ba5ae246a728b25f4316ca2ca"
                    ]
                ],
                "6": [
                    "versions.yml:md5,88a6e42d753077edab8daf829cd4d943"
                ],
                "md_plot": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.mean_difference.png:md5,2fb1e9594c24c0ab21d228e267cb3dcf"
                    ]
                ],
                "model": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.model.txt:md5,3b96713b4e3f027b0347859f02a9038d"
                    ]
                ],
                "normalised_counts": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.normalised_counts.tsv:md5,2aa4880ba5ae246a728b25f4316ca2ca"
                    ]
                ],
                "rdata": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.MArrayLM.limma.rds:md5,dea2517c23b7fa990bbc6eb6add2d00e"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.results.tsv:md5,315c3bb68f55ab03b552111fd37af2ca"
                    ]
                ],
                "session_info": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.R_sessionInfo.log:md5,33a080b950790446e947baa6ddb7051e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,88a6e42d753077edab8daf829cd4d943"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-10-23T11:27:39.844935539"
    },
    "test_limma_differential - voom": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.results.tsv:md5,1e2678fa7304f7f65b6ec646f40175b8"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.mean_difference.png:md5,92e4adebc26a9153edbb165ae69f7fcf"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.MArrayLM.limma.rds:md5,fbaff9956df0ab8e69be07dfe9958058"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.model.txt:md5,d2113d82b76046c319e6602da2ad74d6"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.R_sessionInfo.log:md5,33a080b950790446e947baa6ddb7051e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.normalised_counts.tsv:md5,2aa4880ba5ae246a728b25f4316ca2ca"
                    ]
                ],
                "6": [
                    "versions.yml:md5,88a6e42d753077edab8daf829cd4d943"
                ],
                "md_plot": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.mean_difference.png:md5,92e4adebc26a9153edbb165ae69f7fcf"
                    ]
                ],
                "model": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.model.txt:md5,d2113d82b76046c319e6602da2ad74d6"
                    ]
                ],
                "normalised_counts": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.normalised_counts.tsv:md5,2aa4880ba5ae246a728b25f4316ca2ca"
                    ]
                ],
                "rdata": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.MArrayLM.limma.rds:md5,fbaff9956df0ab8e69be07dfe9958058"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.results.tsv:md5,1e2678fa7304f7f65b6ec646f40175b8"
                    ]
                ],
                "session_info": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.R_sessionInfo.log:md5,33a080b950790446e947baa6ddb7051e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,88a6e42d753077edab8daf829cd4d943"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-10-23T11:27:51.633133328"
    },
    "test_limma_differential - subset_to_contrast": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.results.tsv:md5,70ca2d581759a263ef87a468f8877062"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.mean_difference.png:md5,eab90b40432a0bf832a7aa1226836077"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.MArrayLM.limma.rds:md5,8e1fb2bb5484b95074c315b531e4ac19"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.model.txt:md5,d2113d82b76046c319e6602da2ad74d6"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.R_sessionInfo.log:md5,33a080b950790446e947baa6ddb7051e"
                    ]
                ],
                "5": [

                ],
                "6": [
                    "versions.yml:md5,88a6e42d753077edab8daf829cd4d943"
                ],
                "md_plot": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.mean_difference.png:md5,eab90b40432a0bf832a7aa1226836077"
                    ]
                ],
                "model": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.model.txt:md5,d2113d82b76046c319e6602da2ad74d6"
                    ]
                ],
                "normalised_counts": [

                ],
                "rdata": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.MArrayLM.limma.rds:md5,8e1fb2bb5484b95074c315b531e4ac19"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.results.tsv:md5,70ca2d581759a263ef87a468f8877062"
                    ]
                ],
                "session_info": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.R_sessionInfo.log:md5,33a080b950790446e947baa6ddb7051e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,88a6e42d753077edab8daf829cd4d943"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-10-23T11:27:28.30217231"
    },
    "test_limma_differential - exclude_samples": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.results.tsv:md5,eecb91190030289da100e42247d66eae"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.mean_difference.png:md5,572b313f14f79e6d3760e3fa1a0d399e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.MArrayLM.limma.rds:md5,f66fbaeb4a2df25812fdda407f454243"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.model.txt:md5,d2113d82b76046c319e6602da2ad74d6"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.R_sessionInfo.log:md5,33a080b950790446e947baa6ddb7051e"
                    ]
                ],
                "5": [

                ],
                "6": [
                    "versions.yml:md5,88a6e42d753077edab8daf829cd4d943"
                ],
                "md_plot": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.mean_difference.png:md5,572b313f14f79e6d3760e3fa1a0d399e"
                    ]
                ],
                "model": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.model.txt:md5,d2113d82b76046c319e6602da2ad74d6"
                    ]
                ],
                "normalised_counts": [

                ],
                "rdata": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.MArrayLM.limma.rds:md5,f66fbaeb4a2df25812fdda407f454243"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.results.tsv:md5,eecb91190030289da100e42247d66eae"
                    ]
                ],
                "session_info": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.R_sessionInfo.log:md5,33a080b950790446e947baa6ddb7051e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,88a6e42d753077edab8daf829cd4d943"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-10-23T11:27:16.718342041"
    },
    "test_limma_differential - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.mean_difference.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.MArrayLM.limma.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.model.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.R_sessionInfo.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.normalised_counts.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    "versions.yml:md5,88a6e42d753077edab8daf829cd4d943"
                ],
                "md_plot": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.mean_difference.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "model": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.model.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "normalised_counts": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.normalised_counts.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "rdata": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.MArrayLM.limma.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "session_info": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.R_sessionInfo.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,88a6e42d753077edab8daf829cd4d943"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-10-23T11:28:00.949335071"
    }
}
